src/utils/defer.js

```diff
@@ -1,25 +1,28 @@
-/* flow */
+/* @flow */
 
-export type deferred = {
-  resolve: (result: any) => void,
-  reject: (result: any) => void,
-  promise: Promise
+declare var Resolve: (result: any) => void;
+declare var Reject: (result: any) => void;
+
+export type Defer = {
+  resolve: Resolve,
+  reject: Reject,
+  promise: Promise<any>
 };
 
-function defer(): deferred {
-  let resolve: (result: any) => void;
-  let reject: (result: any) => void;
-  let promise = new Promise(function(
-    innerResolve: (result: any) => void,
-    innerReject: (result: any) => void
+function defer(): Defer {
+  let resolve: Resolve; // eslint-disable-line no-unused-vars
```

Comment:
I was getting this error for line 13 and 14 from lint check, so I used `// eslint-disable-line no-unused-vars`.
<img width="538" alt="screen shot 2017-01-22 at 12 04 34 pm" src="https://cloud.githubusercontent.com/assets/4562118/22179922/d8144366-e09b-11e6-9461-4930e0ff7b68.png">